src/main/java/org/owasp/esapi/Encoder.java

```diff
@@ -365,171 +400,173 @@
-	 * @see <a href="http://www.ibm.com/developerworks/xml/library/x-xpathinjection.html">XPath Injection [ibm.com]</a>
-	 * @see <a href="http://www.packetstormsecurity.org/papers/bypass/Blind_XPath_Injection_20040518.pdf">Blind XPath Injection [packetstormsecurity.org]</a>
-	 *  
-	 * @param input
-	 *      the text to encode for XPath
-	 * @return 
-	 * 		input encoded for use in XPath
-	 */
-	String encodeForXPath(String input);
+    /**
+     * Encode data for use in an XPath query.
+     * 
+     * NB: The reference implementation encodes almost everything and may over-encode. 
+     * 
+     * The difficulty with XPath encoding is that XPath has no built in mechanism for escaping
+     * characters. It is possible to use XQuery in a parameterized way to
+     * prevent injection. 
+     * 
+     * For more information, refer to <a
+     * href="http://www.ibm.com/developerworks/xml/library/x-xpathinjection.html">this
+     * article</a> which specifies the following list of characters as the most
+     * dangerous: ^&"*';<>(). <a
+     * href="http://www.packetstormsecurity.org/papers/bypass/Blind_XPath_Injection_20040518.pdf">This
+     * paper</a> suggests disallowing ' and " in queries.
+     * 
+     * @see <a href="http://www.ibm.com/developerworks/xml/library/x-xpathinjection.html">XPath Injection [ibm.com]</a>
+     * @see <a href="http://www.packetstormsecurity.org/papers/bypass/Blind_XPath_Injection_20040518.pdf">Blind XPath Injection [packetstormsecurity.org]</a>
+     *  
+     * @param input
+     *      the text to encode for XPath
+     * @return 
+     *      input encoded for use in XPath
+     */
+    String encodeForXPath(String input);
 
-	/**
-	 * Encode data for use in an XML element. The implementation should follow the <a
-	 * href="http://www.w3schools.com/xml/xml_encoding.asp">XML Encoding
-	 * Standard</a> from the W3C.
-	 * <p>
-	 * The use of a real XML parser is strongly encouraged. However, in the
-	 * hopefully rare case that you need to make sure that data is safe for
-	 * inclusion in an XML document and cannot use a parse, this method provides
-	 * a safe mechanism to do so.
-	 * 
-	 * @see <a href="http://www.w3schools.com/xml/xml_encoding.asp">XML Encoding Standard</a>
-	 * 
-	 * @param input
-	 * 			the text to encode for XML
-	 * 
-	 * @return
-	 *			input encoded for use in XML
-	 */
-	String encodeForXML(String input);
+    /**
+     * Encode data for use in an XML element. The implementation should follow the <a
+     * href="http://www.w3schools.com/xml/xml_encoding.asp">XML Encoding
+     * Standard</a> from the W3C.
+     * <p>
+     * The use of a real XML parser is strongly encouraged. However, in the
+     * hopefully rare case that you need to make sure that data is safe for
+     * inclusion in an XML document and cannot use a parse, this method provides
+     * a safe mechanism to do so.
+     * 
+     * @see <a href="http://www.w3schools.com/xml/xml_encoding.asp">XML Encoding Standard</a>
```

Comment:
Given my criticism of w3schools in the list email, ought we change this to the RFC standard or do we have some space where I could perhaps discuss this and related topics for reference here?  This week at work I got hit with a vendor trying to squeak by saying "*" was OK in a URI because the URI spec said so... when the real world is a bit more complicated.  There's a spec for the URI, and another spec for the HTTP protocol... I can see this mistake being repeated MANY times.

Reply:
Yep; I changed some of them, but looks like I missed this one. Good eye.

Reply:
Ah, never mind. I did originally miss it, but then later corrected that one. (Note the 'Outdated' label next to the file name.)